STUDIO/org.openl.rules.project/src/org/openl/rules/project/xml/XmlProjectDescriptorSerializer.java

```diff
@@ -27,81 +26,65 @@
      *
      */
     public XmlProjectDescriptorSerializer() {
-        xstream = new XStream(new DomDriver());
-        xstream.addPermission(NoTypePermission.NONE);
-        xstream.allowTypeHierarchy(Module.class);
-        xstream.allowTypeHierarchy(ProjectDescriptor.class);
-        xstream.allowTypeHierarchy(ProjectDependencyDescriptor.class);
-        xstream.allowTypeHierarchy(PathEntry.class);
-        xstream.allowTypeHierarchy(String.class);
-
-        xstream.ignoreUnknownElements();
-        xstream.omitField(ProjectDescriptor.class, "id"); // This field was deprecated
-        xstream.omitField(ProjectDescriptor.class, "log");
-        xstream.omitField(ProjectDescriptor.class, "classLoader");
-        xstream.omitField(ProjectDescriptor.class, "classPathUrls");
-        xstream.omitField(ProjectDescriptor.class, "projectFolder");
-        xstream.omitField(Module.class, "properties"); // properties does not supported by rules.xml
-        xstream.omitField(Module.class, "wildcardName"); // runtime properties
-        xstream.omitField(Module.class, "wildcardRulesRootPath"); // runtime properties
-        xstream.omitField(Module.class, "project"); // runtime properties
-
-        xstream.setMode(XStream.NO_REFERENCES);
-
-        xstream.aliasType(PROJECT_DESCRIPTOR_TAG, ProjectDescriptor.class);
-        xstream.aliasType(MODULE_TAG, Module.class);
-        xstream.aliasType(DEPENDENCY_TAG, ProjectDependencyDescriptor.class);
-        xstream.aliasType(WEBSTUDIO_CONFIGURATION, WebstudioConfiguration.class);
-        xstream.aliasType(PATH_TAG, PathEntry.class);
-        xstream.addImplicitArray(ProjectDescriptor.class, "propertiesFileNamePatterns", PROPERTIES_FILE_NAME_PATTERN);
-        xstream.aliasField(PROPERTIES_FILE_NAME_PROCESSOR, ProjectDescriptor.class, "propertiesFileNameProcessor");
-        xstream.addDefaultImplementation(HashSet.class, Collection.class);
-        xstream.alias("value", String.class);
-
-        xstream.useAttributeFor(PathEntry.class, "path");
-        xstream.aliasField("rules-root", Module.class, "rulesRootPath");
-        xstream.aliasField(METHOD_FILTER_TAG, Module.class, "methodFilter");
-        xstream.registerConverter(new StringValueConverter());
-
-        xstream.aliasField("openapi", ProjectDescriptor.class, "openapi");
-        xstream.aliasField("model-module-name", OpenAPI.class, "modelModuleName");
-        xstream.aliasField("algorithm-module-name", OpenAPI.class, "algorithmModuleName");
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(ProjectDescriptor.class);
+            jaxbMarshaller = jaxbContext.createMarshaller();
+            jaxbMarshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            jaxbMarshaller.setProperty(Marshaller.JAXB_FRAGMENT, true); // don't include header
+            jaxbUnmarshaller = jaxbContext.createUnmarshaller();

```

Comment:
Code duplication across multiple classes. Is it reasonable to extract or not?

Reply:
Yes, we can extract this code to separate class, JAXBSerializer for example